src/ui/steps.py

```
@@ -4,59 +4,108 @@
+    Args:
+        uploaded: The uploaded file object.
+
+    Returns:
+        tuple[bool, str]: Success flag and content-based signature.
+    """
+    try:
+        ext = uploaded.name.split(".")[-1].lower()
+        if ext == "csv":
+            df_new = pd.read_csv(uploaded)
+        else:
+            df_new = pd.read_excel(uploaded)
```

Ruff (0.15.12)

[warning] Use ternary operator `df_new = pd.read_csv(uploaded) if ext == "csv" else pd.read_excel(uploaded)` instead of `if`-`else`-block

Replace `if`-`else`-block with `df_new = pd.read_csv(uploaded) if ext == "csv" else pd.read_excel(uploaded)`

(SIM108)